libs/designsystem/src/lib/components/calendar/calendar.component.ts

```diff
@@ -371,18 +405,16 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
     this.calendarHelper.setSelectedDay(newDate.getDate());
   }
 
-  onDateSelected(newDay: CalendarCell) {
+  _onDateSelected(newDay: CalendarCell) {
     if (newDay.isSelectable && newDay.date) {
-      const newDate = this.activeMonth
-        .clone()
-        .date(newDay.date)
-        .toDate();
+      let newDate = new Date(this.activeMonth);
 
-      // emit equivalent date in utc midnight if timezone is not local
-      const dateToEmit =
-        this.timezone === 'local'
-          ? newDate
-          : moment.utc(moment(newDate).format('YYYY-MM-DD')).toDate();
+      if (this.timezone === 'UTC') {
+        newDate = zonedTimeToUtc(this.subtractTimezoneOffset(newDate), this.timeZoneName);
```

Comment:
Is this safe? 🤔 
In the previous implementation there was a difference between the date used for comparing against the previous selected date (`if (this.hasDateChanged(newDate, this._selectedDate))`) and the dateToEmit...

Reply:
Should be safe enough. Difference is only in the UTC conversion (comparison was previously made on non-converted date). But I explicitly set the new date after the UTC conversion, so the date comparison should be safe

Reply:
I was asked to take a look and I agree with the reasoning. Whichever offset is used for `newDate` (whether local or UTC), the same offset will be used for `this._selectedDate` and I thus believe the comparison to be safe.